huaweicloud/services/iotda/resource_huaweicloud_iotda_batchtask.go

```diff
@@ -0,0 +1,442 @@
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status_desc": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"task_progress": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"total": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"success": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"fail": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"task_details": {
+				Type:      schema.TypeList,
+				Sensitive: true,
+				Computed:  true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"target": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"output": {
+							Type:      schema.TypeString,
+							Sensitive: true,
+							Computed:  true,
+						},
+						"error": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"error_code": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"error_msg": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceBatchTaskCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	isDerived := withDerivedAuth(cfg, region)
+	client, err := cfg.HcIoTdaV5Client(region, isDerived)
+	if err != nil {
+		return diag.Errorf("error creating IoTDA v5 client: %s", err)
+	}
+
+	// First, determine whether to upload batch task files.
+	var targetFileId *string
+	if v, ok := d.GetOk("targets_file"); ok && v.(string) != "" {
+		file, openErr := os.Open(v.(string))
+		if openErr != nil {
+			return diag.Errorf("error opening batch task file: %s", openErr)
+		}
+		defer file.Close()
+
+		uploadOpts := buildBatchTaskFileUploadParams(file)
+		uploadTaskFileResp, uploadErr := client.UploadBatchTaskFile(uploadOpts)
+		if uploadErr != nil {
+			return diag.Errorf("error uploading IoTDA batch task file: %s", uploadErr)
+		}
+
+		if uploadTaskFileResp == nil || uploadTaskFileResp.FileId == nil {
+			return diag.Errorf("error uploading IoTDA batch task file: ID is not found in API response")
+		}
+
+		targetFileId = uploadTaskFileResp.FileId
+	}
+
+	createOpts := buildBatchTaskCreateParams(d, targetFileId)
+	createTaskResp, createTaskErr := client.CreateBatchTask(createOpts)
+	if createTaskErr != nil {
+		// When creating the batch task fails, it is necessary to delete the uploaded task file.
+		deleteTaskFileErr := err
```

Comment:
define the error will be more clear
```suggestion
		var deleteTaskFileErr error
```

Reply:
fixed.